packages/lesswrong/components/tagging/TagPreview.tsx

```diff
@@ -107,11 +125,40 @@ const styles = defineStyles('TagPreview', (theme: ThemeType) => ({
       borderLeft: 'none',
     },
   },
-  descriptionTop: {
+  description: {
     ...(isFriendlyUI && { marginTop: 16 }),
   },
 }));
 
+/*
+/* If the text displayed on hover preview does containt the tag name in the first 100 characters, we use this flag to display a title.
+/* If summaries are present, we must check for the tag name in the first summary, otherwise we use the tag name 
+/* from the main description.
+*/
+const tagShowTitle = (tag: (TagPreviewFragment | TagSectionPreviewFragment) & { summaries?: MultiDocumentEdit[] }) => {
+  if (isFriendlyUI) {
+    return false;
+  }
+
+  const firstSummaryText = tag.summaries?.[0]?.contents?.html
+  const highlightText = getTagDescriptionHtmlHighlight(tag);
+  const openingText: string | undefined = firstSummaryText ?? highlightText;
+
+  if (!openingText) {
+    return true;
+  }
+
+  if (tag.name.length > 100) {
+    return true;
+  }
+
+  // Remove non-word characters and ignore case
+  const openingTextLower = openingText.toLowerCase().replace(/[^\w\s]/g, '');
+  const tagNameLower = tag.name.toLowerCase().replace(/[^\w\s]/g, '');
+
+  return !openingTextLower.slice(0, 100).includes(tagNameLower);
```

Comment:
I don't think we should constrain this to the first 100 characters of the description/summary, maybe we should use the same truncation library that we use for the preview content to check the first sentence?